aclk/https_client.c

```diff
@@ -100,11 +142,104 @@ static int parse_http_hdr(rbuf_t buf, http_parse_ctx *parse_ctx)
     for (ptr = buf_key; *ptr; ptr++)
         *ptr = tolower(*ptr);
 
-    process_http_hdr(parse_ctx, buf_key, buf_val);
+    if (process_http_hdr(parse_ctx, buf_key, buf_val))
+        return 1;
 
     return 0;
 }
 
+static inline void chunked_response_buffer_grow_by(http_parse_ctx *parse_ctx, size_t size)
+{
+    if (unlikely(parse_ctx->chunked_response_size == 0)) {
+        parse_ctx->chunked_response = mallocz(size);
+        parse_ctx->chunked_response_size = size;
+        return;
+    }
+    parse_ctx->chunked_response = reallocz((void *)parse_ctx->chunked_response, parse_ctx->chunked_response_size + size);
+    parse_ctx->chunked_response_size += size;
+}
+
+static int process_chunked_content(rbuf_t buf, http_parse_ctx *parse_ctx)
+{
+    int idx;
+    size_t bytes_to_copy;
+
+    do {
+        switch (parse_ctx->chunked_content_state) {
+            case CHUNKED_CONTENT_CHUNK_SIZE:
+                if (!rbuf_find_bytes(buf, HTTP_LINE_TERM, strlen(HTTP_LINE_TERM), &idx)) {
+                    if (rbuf_bytes_available(buf) >= rbuf_get_capacity(buf))
+                        return PARSE_ERROR;
+                    return NEED_MORE_DATA;
+                }
+                if (idx == 0) {
+                    parse_ctx->chunked_content_state = CHUNKED_CONTENT_FINAL_CRLF;
+                    continue;
+                }
+                if (idx >= HTTP_HDR_BUFFER_SIZE) {
+                    netdata_log_error("Chunk size is too long");
+                    return PARSE_ERROR;
+                }
+                char buf_size[HTTP_HDR_BUFFER_SIZE];
+                rbuf_pop(buf, buf_size, idx);
+                buf_size[idx] = 0;
+                parse_ctx->chunk_size = strtol(buf_size, NULL, 16);
```

Comment:
Is there a specific reason to use `strtol` instead our [str2l](https://github.com/netdata/netdata/blob/483d8481a5a5edf72630068534feb1a4d228ed0b/libnetdata/inlined.h#L149-L158)?

Reply:
base 16 + some extra checks that strtol does